c_src/sqlite3_nif.c

```diff
@@ -628,30 +615,49 @@ exqlite_multi_step(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
         return make_error_tuple(env, am_invalid_chunk_size);
     }
 
+    int column_count = sqlite3_column_count(statement->statement);
+
+    ERL_NIF_TERM* cells = NULL;
+    cells               = enif_alloc(sizeof(ERL_NIF_TERM) * column_count);
+
+    if (!cells) {
```

Comment:
```suggestion
    if (!cells && column_count > 0) {
```

Something like this?

Reply:
Yea I think that would work. It would be safe because the list of results returned would be an empty array.